rye/src/cli/rye.rs

```diff
@@ -641,32 +633,40 @@ fn add_rye_to_path(mode: &InstallMode, shims: &Path, ask: bool) -> Result<(), Er
             );
             echo!("It is highly recommended that you add it.");
 
+            use whattheshell::Shell;
```

Comment:
I don't think it's a good idea to only write the profile file depending on the current shell. There are lots of situations where people run multiple shells. That's for instance also why fish is shown in addition instead of instead. If we do want to write this `.zprofile` file then I think it should be done based on the mere presence of zsh (`which` finds it) and not if it's the active shell.

Reply:
There is another issue with this that I noticed: It does not work with the suggested install workflow (`curl https://rye-up.com/get | bash`). If the script is piped into bash, `whattheshell` will detect bash and still use `.profile` even though the default login shell is zsh. And it sounds very unintuitive to me to pipe the script into zsh.

I think we can detect if zsh is installed via `which`, but this does not tell us if zsh is acually used. Also, as you @mitsuhiko pointed out, `.profile` is loaded from zsh on your machine, while it is clearly not on my machine. So there is some ambiguity on the usage of `.profile`.

Under these circumstances, I wonder if automatically adding rye to `.zprofile` should be the way to go at all. Maybe a note that the user might have to add it manually is a better approach - similar with the note rye already has for fish. This note can then be triggered by the presence of zsh on the system.

What do you think?

Reply:
Depending solely on the current shell for writing the profile file indeed seems problematic, especially considering the variety of shell setups users might have. 

I'm going to close this PR. Thank you for your insightful input.